internal/app/handle.go

```diff
@@ -0,0 +1,75 @@
+package app
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type fighter struct {
+	Name string `json:"name"`
+}
+
+// struct to hold the new concatination of the user's name
+type allegiance struct {
+	Message string `json:"message"`
+}
+
+// gokuHandler accepts a post request that takes the user's name from the JSON request
+// and responds back with that user's name + some message
+func gokuHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodPost:
+		// Reads the body of the JSON request into the fighter struct
```

Comment:
functions and methods like this can get pretty crazy. i recommend separating all of these cases out as functions. and calling the functions. `handleGokuPost()`, `handleGokuDefault()`, etc. this will be useful when you move onto the next tasks with more complicated logic.